new guide setting-url-wordpress-source-data (#274)
frontity/docs: docs/guides/setting-url-wordpress-source-data..md

> # Setting the URL of the WordPress source of data

Reviewer: Replace
"Setting the URL of the WordPress source of data"
with
"Setting the URL of the WordPress data source"

The most important setting in a Frontity project is the WordPress installation that can be used as the source of data

> The format of the WordPress URL will be different depending on the type of WordPress installation used with the Frontity project so the type of WordPress installation may require different ways of setting this url in the Frontity settings

Reviewer: Replace
"The format of the WordPress URL will be different depending on the type of WordPress installation used with the Frontity project so the type of WordPress installation may require different ways of setting this url in the Frontity settings"
with
"The format of the WordPress URL varies depending on the type of WordPress installation used as the data source for the Frontity project, so the type of WordPress installation determines how this the URL should be set in the Frontity configuration file `frontity.settings.js`"

> The main properties in the settings of this package that are implied in the URL of the WordPress source of data are:

Reviewer: Replace
"The main properties in the settings of this package that are implied in the URL of the WordPress source of data are:"
with
"The main properties in the settings of this package where the values are determined by the type of WordPress installation used as the source of data are:"

> - `state.source.url`:  The URL of the WordPress
> - `state.source.api`:  The URL of the WordPress REST API
> - `state.wpSource.isWpCom`:  A flag to indicate a special use case of wordpress.com sites (Personal or Premium plans)
> 
> `state.source.api` can be set manually but it can be computed from the value of `state.source.url` in most of the cases

Reviewer 2: I would remove the `state.source.api` mentions. **It should never be set manually by the end-users from now on.**

I would add `state.wpSource.prefix` because that one can be changed by the end-users if they use something different than `/wp-json`.

I would also add info about the default values and requirements.

```suggestion
- `state.source.url`:  The URL of the WordPress. Required.
- `state.wpSource.isWpCom`:  A flag to indicate a special use case of wordpress.com sites (Personal or Premium plans). It is not required for Free wordpress.com sites. Defaults to `false`.
- `state.wpSource.prefix`: The prefix of the API. Defaults to `/wp-json`. It is not used if `isWpCom` is `true`.
```

## Wordpress scenarios

### A self-hosted WordPress site 

> Most of the Frontity projects will use a custom domain for a self-hosted WordPress site ([wp.org](http://wp.org/)) like for example `https://test.frontity.org/`

Reviewer: Replace
"Most of the Frontity projects will use a custom domain for a self-hosted WordPress site ([wp.org](http://wp.org/)) like for example `https://test.frontity.org/`"
with
"Most Frontity projects will use a self-hosted WordPress site ([wp.org](http://wp.org/)) with a custom domain, such as, for example, `https://test.frontity.org/`."

> The recommended way for setting this URL is through the `state.source,url`

Reviewer 2: ```suggestion
The recommended way for setting this URL is through the `state.source.url`
```

> The recommended way for setting this URL is through the `state.source,url`

Reviewer: Replace
"The recommended way for setting this URL is through the `state.source,url`"
with
"The recommended way of setting this URL is via the `state.source.url` property."

```javascript
// frontity.settings.js
export default {
  packages: [
    {
      name: "@frontity/wp-source",
      state: {
        source: {
          url: "https://test.frontity.org",
        },
      },
    },
  ],
};
```

In this example the computed values would be:

> - `state.source.api`:  `https://test.frontity.org/wp-json` (value computed from `state.source.url`)

Reviewer 2: We can say here that `state.source.api` is computed from `state.source.url` + `state.wpSource.prefix`.

```suggestion
- `state.source.api`:  `https://test.frontity.org/wp-json` (value computed from `state.source.url` and `state.wpSource.prefix`).
```

- `state.wpSource.isWpCom`: `false` (value derived `state.source.api`)

The same recommendation applies for custom domains used with a [Business wordpress.com plan](https://wordpress.com/support/business-plan/) 

### A Free wordpress.com plan 

> Some other Frontity projects will use a free wordpress.com installation with URL's pointing to subdomains of wordpress.com like for example  `https://frontitytest.wordpress.com/` 

Reviewer: Replace
"Some other Frontity projects will use a free wordpress.com installation with URL's pointing to subdomains of wordpress.com like for example  `https://frontitytest.wordpress.com/`"
with
"Some Frontity projects may use a free wordpress.com installation with the URL pointing to a subdomain of wordpress.com, such as: `https://frontitytest.wordpress.com/`."

> The recommended way for setting this URL is also through the `state.source,url`

Reviewer 2: ```suggestion
The recommended way for setting this URL is also through the `state.source.url`
```

> The recommended way for setting this URL is also through the `state.source,url`

Reviewer: Replace
"The recommended way for setting this URL is also through the `state.source,url`"
with
"The recommended way of setting this URL is also via the `state.source.url` property."

```javascript
// frontity.settings.js
export default {
  packages: [
    {
      name: "@frontity/wp-source",
      state: {
        source: {
          url: "https://frontitytest.wordpress.com/",
        },
      },
    },
  ],
};
```

In this example the computed values would be:
- `state.source.api`:  `https://frontitytest.wordpress.com//wp-json` (value computed from `state.source.url`)
- `state.wpSource.isWpCom`: `true` (value derived `state.source.api`)

### A Personal or Premium wordpress.com plan 

> A less frequent use case is using a Personal or Premium wordpress.com. These plans allows you to use a custom domain but this a special case because the final REST API url is available through a special URL

Reviewer: Replace
"A less frequent use case is using a Personal or Premium wordpress.com. These plans allows you to use a custom domain but this a special case because the final REST API url is available through a special URL"
with
"A less frequent use case is when a Personal or Premium wordpress.com site is used as the data source. These plans allow you to use a custom domain, but in these cases the REST API is available via a different URL format."

> The recommended way for setting this URL is also through the `state.source,url` but specifying it is a Personal or Premium wordpress.com plan installation by setting `state.wpSource.isWpCom` to `true`

Reviewer 2: ```suggestion
The recommended way for setting this URL is also through the `state.source.url` but specifying it is a Personal or Premium wordpress.com plan installation by setting `state.wpSource.isWpCom` to `true`
```

> The recommended way for setting this URL is also through the `state.source,url` but specifying it is a Personal or Premium wordpress.com plan installation by setting `state.wpSource.isWpCom` to `true`

Reviewer: Replace
"The recommended way for setting this URL is also through the `state.source,url` but specifying it is a Personal or Premium wordpress.com plan installation by setting `state.wpSource.isWpCom` to `true`"
with
"The recommended way of setting this URL is also via the `state.source.url` property, but in addition to this you also need to specify that it is a Personal or Premium wordpress.com plan installation by setting `state.wpSource.isWpCom` to `true`."

```javascript
// frontity.settings.js
export default {
  packages: [
    {
      name: "@frontity/wp-source",
      state: {
        source: {
          url: "https://test-premium-plan.frontity.org",
        },
        wpSource: {
          isWpCom: true
        } 
      },
    },
  ],
};
```

In this example the computed values would be:
- `state.source.api`: ` https://public-api.wordpress.com/wp/v2/sites/test-premium-plan.frontity.org` (value computed from `state.source.url` & `state.wpSource.isWpCom`)

---

> All these scenarios and their different combinations depending on how you set the values are described in detail in the following table

Reviewer 2: I would not include this table. It was made for internal purposes when we needed to work on the internal logic of these props. Also, it could be a bit misleading because nobody should use `state.source.api` anymore.

I think the written explanations are super clear so I would only leave those 🙂
